src/Discord.Net.Commands/Dependencies/DependencyMap.cs

```diff
@@ -38,6 +38,8 @@ public bool TryAddTransient<T>() where T : class, new()
         public void AddFactory<T>(Func<T> factory) where T : class
         {
             var t = typeof(T);
+            if (typeof(T) == typeof(IDependencyMap) || typeof(T) == typeof(CommandService))
+                throw new InvalidOperationException("The dependency map cannot contain services directly added as IDependencyMap or CommandService. Only Implimentations and Derivatives are permitted");
```

Comment:
I'd clean this up by looking for each type individually and splitting this into two exception messages.